bcl2fastq/index_summary.py

```diff
@@ -89,14 +111,17 @@ def update_pruned_unknown_hidden(run_alias):
 )
 def update_known_index_bar(run_json):
     run = pandas.read_json(run_json, orient='split')
-
-    run['library'] = run['SampleID'].str.extract(
-        'SWID_\d+_(\w+_\d+_.*_\d+_[A-Z]{2})_'
+        # convert a Json string into a panda object where orient is the expected format of json 
+    run['library'] = run['SampleID'].str.extract(   # create DataFrame via str.extract 
+        'SWID_\d+_(\w+_\d+_.*_\d+_[A-Z]{2})_' 
     )
+        #replace index library with index sampleID as a string
     run['index'] = run['Index1'].str.cat(
-        run['Index2'].fillna(''), sep=' '
+        run['Index2'].fillna(''), sep=' ' #concatenate the strings with a space in the middle
     )
-
+    run = run.sort_values('SampleNumberReads', ascending=False)
```

Comment:
Why the sort?

The `SampleNumberReads` values are added together at 

https://github.com/oicr-gsi/dashi/blob/294ca750dda09a50824b681cf473e137d79fe32e/bcl2fastq/index_summary.py#L130

and the order should not affect final result.